### PR TITLE
Verify Login test

### PR DESCRIPTION
**Pull request by AdrianKostya**

### CODE REVIEW

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java [hunk 1]
```diff
@@ -79,4 +79,23 @@ public CarinaDescriptionPageBase login() {
...
 
+    @Override
+    public boolean verifyFieldsArePresent() {
+        if (nameInputField.isElementPresent()&&passwordInputField.isElementPresent()
+                && maleRadioBtn.isElementPresent()
+                &&femaleRadioBtn.isElementPresent()
+        ){
+            return true;
+        }
+        return false;
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java]: bad practice
if you need verify a lot of fields use one by one(one method for one field)
Example: isNameFieldPresented, ....

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java [hunk 2]
```diff
@@ -79,4 +79,23 @@ public CarinaDescriptionPageBase login() {
...
+    }
+
+    public boolean isSexSelected(){
+        if(maleRadioBtn.isChecked()||femaleRadioBtn.isChecked()){
+            return true;
+        }
+        return false;
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/LoginPage.java]: see above

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/ios/LoginPage.java [hunk 3]
```diff
@@ -85,4 +85,23 @@ public CarinaDescriptionPageBase login(){
...
+	}
+
+	public boolean isSexSelected(){
+		if(maleRadioBtn.isChecked()||femaleRadioBtn.isChecked()){
+			return true;
+		}
+		return false;
+	}
+
+
```
Comment (vrudyk) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/ios/LoginPage.java]: if you don’t implement for ios use this in methods
private static final String THIS_METHOD_IS_NOT_IMPLEMENTED_FOR_IOS = "This method is not yet implemented for iOS";
throw new UnsupportedOperationException(THIS_METHOD_IS_NOT_IMPLEMENTED_FOR_IOS);

File: src/main/resources/_testdata.properties [hunk 4]
```diff
@@ -1,4 +1,5 @@
 #===============================================================#
 #========================== Test data ==========================#
 #===============================================================#
-test_credentials=test@gmail.com/{crypt:8O9iA4+f3nMzz85szmvKmQ==}
\ No newline at end of file
+test_credentials=test@gmail.com/{crypt:8O9iA4+f3nMzz85szmvKmQ==}
+pass={crypt:OsDABhNRINuvtYfjTWFZnA==}
```
Comment (vrudyk) [inline src/main/resources/_testdata.properties]: revert old lines, don’t use it

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 5]
```diff
@@ -85,4 +86,25 @@ public void testUIElements() {
...
+    @TestLabel(name = "feature", value = {"mobile", "regression"})
+    public void verifyLoginPage (){
+        WelcomePageBase welcomePageBase = initPage(getDriver(), WelcomePageBase.class);
+        Assert.assertTrue(welcomePageBase.isPageOpened(), "Welcome page is not opened");
+        LoginPageBase loginPageBase = welcomePageBase.clickNextBtn();
+        Assert.assertTrue(loginPageBase.verifyFieldsArePresent(), "Some fields is not present");
+        Assert.assertFalse(loginPageBase.isLoginBtnActive(), "Login is active before input fields");
+        String text = "Test user";
+        loginPageBase.typeName(text);
+        loginPageBase.typePassword(R.TESTDATA.get("pass"));
```
Comment (vrudyk) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: create random String and use it, the same as String text

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 6]
```diff
@@ -85,4 +86,25 @@ public void testUIElements() {
...
+    @Test(description = "JIRA#DEMO-0011")
+    @MethodOwner(owner = "akostya")
+    @TestLabel(name = "feature", value = {"mobile", "regression"})
+    public void verifyLoginPage (){
+        WelcomePageBase welcomePageBase = initPage(getDriver(), WelcomePageBase.class);
+        Assert.assertTrue(welcomePageBase.isPageOpened(), "Welcome page is not opened");
+        LoginPageBase loginPageBase = welcomePageBase.clickNextBtn();
+        Assert.assertTrue(loginPageBase.verifyFieldsArePresent(), "Some fields is not present");
+        Assert.assertFalse(loginPageBase.isLoginBtnActive(), "Login is active before input fields");
+        String text = "Test user";
```
Comment (vrudyk) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: its user name
rename string object

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 7]
```diff
@@ -85,4 +86,25 @@ public void testUIElements() {
...
+        loginPageBase.typeName(text);
+        loginPageBase.typePassword(R.TESTDATA.get("pass"));
+        loginPageBase.selectMaleSex();
+        Assert.assertTrue(loginPageBase.isSexSelected());
```
Comment (vrudyk) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: add message

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/AndroidLoginPage.java [hunk 8]
```diff
@@ -1,5 +1,6 @@
 package com.qaprosoft.carina.demo.mobile.gui.pages.android;
 
+import com.qaprosoft.carina.core.foundation.utils.R;
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/AndroidLoginPage.java]: create import list with this rules:

import java
import org
import com

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/AndroidLoginPage.java [hunk 9]
```diff
@@ -79,4 +87,52 @@ public CarinaDescriptionPageBase login() {
         return clickLoginBtn();
     }
 
+
+    public boolean isNameInputFieldArePresent() {
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/AndroidLoginPage.java:90]: Bad practice to implement methods for this way

Better way:
   public boolean isNameInputFieldArePresent() {
        return nameInputField.isElementPresent();
}

Implement all methods with this logic with this way

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/ios/IOSLoginPage.java [hunk 10]
```diff
@@ -1,5 +1,7 @@
 package com.qaprosoft.carina.demo.mobile.gui.pages.ios;
 
+import com.qaprosoft.carina.core.foundation.utils.R;
+import com.qaprosoft.carina.demo.constant.ProjectConstant;
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/ios/IOSLoginPage.java]: change import list

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 11]
```diff
@@ -1,7 +1,11 @@
 package com.qaprosoft.carina.demo;
 
+import com.qaprosoft.carina.core.foundation.crypto.CryptoConsole;
+import com.qaprosoft.carina.core.foundation.crypto.CryptoTool;
+import com.qaprosoft.carina.core.foundation.utils.R;
```
Comment (MuslimSolvd) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java]: change import list

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/AndroidLoginPage.java [hunk 12]
```diff
@@ -42,6 +43,13 @@ public void typeName(String name) {
...
 
+    public String getName (){
+        return nameInputField.getText();
+    }
+
+    public String getPassword(){
+        return passwordInputField.getText();
+    }
     @Override
     public void typePassword(String password) {
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/AndroidLoginPage.java:54]: Bad practice to implement methods with mix logic

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/common/LoginPageBase.java [hunk 13]
```diff
@@ -1,5 +1,6 @@
 package com.qaprosoft.carina.demo.mobile.gui.pages.common;
 
+import com.qaprosoft.carina.core.foundation.utils.R;
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/common/LoginPageBase.java]: change import list

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/AndroidCarinaDescriptionPage.java [hunk 14]
```diff
@@ -31,10 +32,17 @@
...
 
-    public CarinaDescriptionPage(WebDriver driver) {
+    @FindBy(className = "android.webkit.WebView")
+    private ExtendedWebElement mapLinkClass;
+
+    public AndroidCarinaDescriptionPage(WebDriver driver) {
         super(driver);
     }
 
+    public void clickOnBurgerBtn() {
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/AndroidCarinaDescriptionPage.java]: better name of method - openHamburgerMenu

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/AndroidLoginPage.java [hunk 15]
```diff
@@ -79,4 +87,29 @@ public CarinaDescriptionPageBase login() {
...
+
+    public boolean isMaleRadioBtnArePresent() {
+        return maleRadioBtn.isElementPresent();
+    }
+
+    public boolean isFemaleRadioBtnArePresent() {
+        return femaleRadioBtn.isElementPresent();
+    }
+
+    public boolean isMaleSexSelected() {
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/AndroidLoginPage.java]: better name of method - isMaleGenderSelected

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/AndroidLoginPage.java [hunk 16]
```diff
@@ -79,4 +87,29 @@ public CarinaDescriptionPageBase login() {
...
+        return maleRadioBtn.isChecked();
+    }
+
+    public boolean isFemaleSexSelected() {
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/android/AndroidLoginPage.java]: the same

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/common/ChartsPageBase.java [hunk 17]
```diff
@@ -1,7 +1,7 @@
 package com.qaprosoft.carina.demo.mobile.gui.pages.common;
 
-import com.qaprosoft.carina.core.gui.AbstractPage;
 import org.openqa.selenium.WebDriver;
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/common/ChartsPageBase.java:3]: need space between imports

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/common/LoginPageBase.java [hunk 18]
```diff
@@ -24,4 +24,21 @@ public LoginPageBase(WebDriver driver) {
...
 
+	public abstract boolean isNameInputFieldArePresent();
+
+	public abstract boolean isPasswordInputFieldArePresent();
+
+	public abstract boolean isMaleRadioBtnArePresent();
+
+	public abstract boolean isFemaleRadioBtnArePresent();
+
+
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/common/LoginPageBase.java:36]: delete redundant space

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/common/LoginPageBase.java [hunk 19]
```diff
@@ -24,4 +24,21 @@ public LoginPageBase(WebDriver driver) {
...
+	public abstract boolean isMaleSexSelected();
+
+	public abstract boolean isFemaleSexSelected();
+
+	public abstract String getPassword();
+
+
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/common/LoginPageBase.java]: the same

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/ios/IOSCarinaDescriptionPage.java [hunk 20]
```diff
@@ -1,26 +1,30 @@
...
 import com.qaprosoft.carina.core.foundation.webdriver.decorator.annotations.Predicate;
 import com.qaprosoft.carina.demo.mobile.gui.pages.common.*;
-import org.openqa.selenium.WebDriver;
-import org.openqa.selenium.support.FindBy;
 
 @DeviceType(pageType = Type.IOS_PHONE, parentClass = CarinaDescriptionPageBase.class)
-public class CarinaDescriptionPage extends CarinaDescriptionPageBase {
+public class IOSCarinaDescriptionPage extends CarinaDescriptionPageBase {
 
     private static final String THIS_METHOD_IS_DEFINED_ONLY_IN_ANDROID = "This method is not yet implemented for iOS";
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/ios/IOSCarinaDescriptionPage.java]: delete and use constant from class ProjectConstant

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/ios/IOSCarinaDescriptionPage.java [hunk 21]
```diff
@@ -1,26 +1,30 @@
...
     @Predicate
     private ExtendedWebElement webViewTitle;
 
-    public CarinaDescriptionPage(WebDriver driver) {
+    public IOSCarinaDescriptionPage(WebDriver driver) {
         super(driver);
     }
 
+    public  void clickOnBurgerBtn(){
+        throw new UnsupportedOperationException(THIS_METHOD_IS_DEFINED_ONLY_IN_ANDROID);
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/ios/IOSCarinaDescriptionPage.java]: use another present constant from ProjectConstant

File: src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/ios/IOSCarinaDescriptionPage.java [hunk 22]
```diff
@@ -1,26 +1,30 @@
...
+    }
     @Override
     public WebViewPageBase navigateToWebViewPage() {
         throw new UnsupportedOperationException(THIS_METHOD_IS_DEFINED_ONLY_IN_ANDROID);
```
Comment (MuslimSolvd) [inline src/main/java/com/qaprosoft/carina/demo/mobile/gui/pages/ios/IOSCarinaDescriptionPage.java]: the same

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 23]
```diff
@@ -56,7 +57,7 @@ public void testWebView() {
         hideKeyboard();
         contactUsPage.submit();
         Assert.assertTrue(contactUsPage.isSuccessMessagePresent() || contactUsPage.isRecaptchaPresent(),
-            "message was not sent or captcha was not displayed");
+                "message was not sent or captcha was not displayed");
```
Comment (MuslimSolvd) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java:65]: delete spaces

File: src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java [hunk 24]
```diff
@@ -85,4 +86,50 @@ public void testUIElements() {
...
+        CarinaDescriptionPageBase carinaDescriptionPage = loginPage.login();
+        carinaDescriptionPage.isPageOpened();
+        MapsPageBase mapsPageBase = carinaDescriptionPage.navigateToMapPage();
+        softAssert.assertTrue(mapsPageBase.isZoomInPresent(), "Zoom in is not present");
+        softAssert.assertTrue(mapsPageBase.isZoomOutPresent(), "Zoom out is not present");
+        softAssert.assertTrue(mapsPageBase.isZoomInAboveZoomOut(), "Zoom in is not above zoom out");
+        Assert.assertTrue(carinaDescriptionPage.isPageOpened(), "Carina description page is not opened");
+        softAssert.assertAll();
+    }
+
```
Comment (MuslimSolvd) [inline src/test/java/com/qaprosoft/carina/demo/MobileSampleTest.java:138]: delete space

**Pull request merged**